nautobot_golden_config/views.py

```diff
@@ -55,11 +56,25 @@ def extra_context(self):
 
     def alter_queryset(self, request):
         """Build actual runtime queryset as the build time queryset provides no information."""
-        qs = Device.objects.none()
-        for obj in models.GoldenConfigSetting.objects.all():
-            qs = qs | obj.get_queryset().distinct()
-
-        return self.queryset.filter(id__in=qs)
+        params = QueryDict()
+        params._mutable = True  # pylint: disable=protected-access
+        for obj in models.GoldenConfigSetting.objects.values_list("scope", flat=True).distinct():
+
+            params.update(QueryDict(urllib.parse.urlencode(obj, doseq=True)))
+        filtered_query = self.filterset(params, self.queryset)
```

Comment:
Since this block is a bit mysterious to newcomers, can you please describe what is actually happening here for posterity?